tests/test_context_managers.py

```diff
@@ -6,14 +6,19 @@
 
```

Comment:
These should be instance variables. Otherwise you can't call the `exposed_context` method twice (or you have to deal with resetting them after each call but that's worse..).